### PR TITLE
Merge dev into master. openshift-test-pipeline

### PR DESCRIPTION
**Pull request by dbrennand**
This PR adds all of the work to create the openshift-test-pipeline.
Includes:
* Setup and run the test pipeline via [run.py](run.py)
* Run the test pipeline via a webhook trigger.
* Slack notifications + plugins installed.
* Setup secondary pipeline for creating Jenkins slave via docker/dockerfile.
* Readme with instructions on prerequisites and usage of [run.py](run.py)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,74 @@
...
+
+11. Click **Add**
+
+12. Select the `slack-token` from the **Credential** dropdown menu
+
+13. Set the **Default channel / member id** to the channel you wish to send notifications to
+
+14. To see if notifications are working, click **Test Connection**
+
+15. Once you have verified everything if working, click **Save**
```
Comment (benbacon) [inline README.md]:
```suggestion
15. Once you have verified everything is working, click **Save**
```

**Review by benbacon — DISMISSED** [review]
Looks good

File: README.md [hunk 2]
```diff
@@ -71,4 +71,4 @@ A pipeline for running tests against an OpenShift cluster.
 
 14. To see if notifications are working, click **Test Connection**
 
-15. Once you have verified everything if working, click **Save**
+15. Once you have verified everything is working, click **Save**
```
Comment (dbrennand) [inline README.md:74]: Thanks 😄

**Review by benbacon — APPROVED** [review]
Looks good, again

**Comment by dbrennand** [comment]
Thanks merging 😄

**Pull request merged**